quint/src/generated/Quint.g4

```diff
@@ -214,6 +221,21 @@ operator: (AND | OR | IFF | IMPLIES |
 literal: (STRING | BOOL | INT)
         ;
 
+// A (possibly) qualified identifier, like `Foo` or `Foo::bar`
+qualId : IDENTIFIER ('::' IDENTIFIER)* ;
+// An unqualified identifier that raises an error if a qualId is supplied
+simpleId[context: string]
+    : IDENTIFIER
+    | qualId {
+        const err = quintErrorToString(
+          { code: 'QNT008',
+            message: "Identifiers in a " + $context + " cannot be qualified with '::'. Found " + $qualId.text + "."
+          },
+        )
+        this.notifyErrorListeners(err)
```

Comment:
I'm torn whether we should use `quintErrorToString` here at all (with all it's implications, making error fields optional), or simply pass a string as for `QNT006` above. WDYT?

Reply:
So my justification for bringing it in is that we want to put white a few more errors here probably, and using the type and function ensures we are using valid error ids, and if/when we overhaul the error system or add hints to tic etc, we will know to add them here as well. I do agree it’s more overhead, but already in this case it caught a typo on in my error if 😅.

If you feel strongly that we should stick to strings tho I’m ok with that. We could get @bugarela’s opinion as a tiebreaker! :)

regarding making the reference field optional, note that we already effectively treat it as optional (afaict), in that we allow references which aren’t ok our location map and then just print and unknown location.

Reply:
> that we want to put white a few more errors here probably, and using the type and function ensures we are using valid error ids, and if/when we overhaul the error system or add hints to tic etc, we will know to add them here as well

sgtm 👍